ci-operator/step-registry/ipi/install/heterogeneous/ipi-install-heterogeneous-commands.sh

```diff
@@ -6,98 +6,133 @@ set -o pipefail
+  azure_subscription_id=$(yq-v4 .subscriptionId < "$azure_auth_location")
+  az login --service-principal -u "$azure_auth_client_id" -p "$azure_auth_client_secret"\
+    --tenant "$azure_auth_tenant_id" --output none
+  az account set --subscription "${azure_subscription_id}"
+  echo "Setting up the boot image for the ${ADDITIONAL_WORKER_ARCHITECTURE} workers..."
+  vhd_url=$(oc -n openshift-machine-config-operator get configmap/coreos-bootimages -oyaml | \
+    yq-v4 ".data.stream \
+         | eval(.).architectures.${ADDITIONAL_WORKER_ARCHITECTURE}.\"rhel-coreos-extensions\".\"azure-disk\".url")
+  vhd_name=$(basename "${vhd_url}")
+  infra_id=$(yq-v4 '.infraID' < "${SHARED_DIR}"/metadata.json)
+  rg_name="${infra_id}-rg"
+  sa_name=$(az storage account list -g "${rg_name}" | yq-v4 '.[] | select(.name == "cluster*").name')
+  AZURE_STORAGE_KEY=$(az storage account keys list -g "${rg_name}" --account-name "${sa_name}" --query "[0].value" -o tsv)
+  export AZURE_STORAGE_KEY
+  az storage blob copy start --account-name "${sa_name}" \
+      --destination-blob "${vhd_name}" --destination-container vhd --source-uri "$vhd_url"
+  gallery_name=$(az sig list -g "${rg_name}" | yq-v4 '.[].name')
+  image_name="${infra_id}-gen2-${ADDITIONAL_WORKER_ARCHITECTURE}"
+  storage_blob_url=$(az storage blob url --account-name "${sa_name}" --container-name vhd --name "${vhd_name}" -o tsv)
+  az sig image-definition create --resource-group "${rg_name}" --gallery-name "${gallery_name}" \
+    --gallery-image-definition "${image_name}" --publisher "RedHat" --offer "rhcos" \
+    --sku "rhcos-${ADDITIONAL_WORKER_ARCHITECTURE}" --os-type linux --hyper-v-generation V2 \
+    --architecture "$(sed 's/aarch64/Arm64/;s/x86_64/x64/' <<< "${ADDITIONAL_WORKER_ARCHITECTURE}")"
 
-wait "$!"
-ret="$?"
+  region=$(az group show --name "${rg_name}" | yq-v4 '.location')
+  for i in $(seq 1 15) max; do
+    [ "$i" == max ] && { echo "Timeout exceeded while waiting for the VHD blob copy to conclude. Failing..."; exit 3; }
+    sleep 60
+    [ X"$(az storage blob show --container-name vhd --name "${vhd_name}" --account-name "${sa_name}" \
+      -o tsv --query properties.copy.status)" == X"success" ] && break
+    echo "Waiting for the VHD blob copy to conclude... (timeout in $(( 15 - i )) minutes)"
+  done
+  echo "The VHD image is now available. Creating the image version..."
+  az sig image-version create --resource-group "${rg_name}" \
+    --gallery-name "${gallery_name}" --gallery-image-definition "${image_name}" \
+    --gallery-image-version 1.0.0 --target-regions "${region}" \
+    --os-vhd-uri "${storage_blob_url}" --os-vhd-storage-account "${sa_name}"
+  echo "The image version for the ${ADDITIONAL_WORKER_ARCHITECTURE} workers has been created... "
+  echo "Patching the MachineSet..."
+  resource_id="/resourceGroups/${rg_name}/providers/Microsoft.Compute/galleries/${gallery_name}/images/${image_name}/versions/1.0.0"
```

Comment:
you can have the image version be any string and just reference it as "latest"..that way you won't have to set a specific version. see https://github.com/openshift/installer/blob/master/pkg/asset/machines/azure/machines.go#L120

Reply:
done and the version is now set similarly to the one the installer sets: (41x.yz.YYYYMMDD)

https://github.com/openshift/release/pull/35101/files#diff-9e75860c660730e2cb77472d634a02aa6da64e7bac073d6cbfb7daf4133188a1R119